### DISCUSSION TITLE
Add a setting "EON-PS"

### DISCUSSION THREAD

**Discussion by lxsxl** [PE5PVB/TEF6686_ESP32#433] [Ideas]
Two Values:
1. "dynamic" (optimized if dynamic EON-PS are used, will be overwritten in each cycle)
2. "static" (optimized if static EON-PS are used, collects PS fragments with each cycle)

**Comment by PE5PVB** [comment]
How to detect dynamic or static PS?

**Comment by lxsxl** [comment]
Detection of dynamic or static EON-PS is not needed. The idea was to add a setting in the menu how EON-PS shall be treated. Actually it is only improved for „dynamic“ (PS will be deleted and rewritten with each EON cycle). With an additional improvement for „static“ it should keep the already received and add the newly received characters (resp. exchange them as there may have been errors). As static EON-PS is common in Germany (and as far as I know also in Belgium) this would improve EON function especially in case of poor reception quality.

**Reply by PE5PVB** [reply]
That is correct, old EON PS is not deleted but always overwritten. When for example going from RADIO2 to HITS, the chars after HITS are transmitted as space in EON. This is standard PS behaviour because there is no 0x0d specified in RDS for PS.

**Reply by lxsxl** [reply]
Okay, so characters from the first cycle will get lost if they are not received in the second cycle. In the expected "static" mode it should keep the characters from the previous cycle(s) as long as the respective block(s) is/are not received again.

**Reply by PE5PVB** [reply]
No nothing will get lost. Only overwritten what is received

**Reply by lxsxl** [reply]
Just checked it:
First cycle, EON-PS "WDxxxxxx" is displayed, in the next cycle it is "xxxxxxxx", next cycle "WDR_xxxx" and so on... So I assuem that the PS-pairs are not collected.

**Comment by PE5PVB** [comment]
I changed the way of parsing EON PS to prevent this.

**Comment by lxsxl** [comment]
Now EON-PS are only displayed if they have been received completely. It would be better if also fragments are displayed and later received parts should be added step by step (better a few characters in case of poor reception quality!). But anyway: with the new behaviour there is no need for a setting "dynamic/static" anymore.

**Comment by PE5PVB** [comment]
OK. Changed it that also incomplete PS fragments are shown.

**Comment by lxsxl** [comment]
It seems to work fine now. I close this discussion as no further developement is needed for this intention.
